### PR TITLE
Stat Tracker additions 

### PR DESCRIPTION
**Pull request by ethanb1145**
## Description

Added remaining game methods and league methods, made fixes to league class.

## Type of change

- [x] fix
- [x] feat
- [x] test
- [ ] refactor
- [ ] docs

## Checklist

- [x] code has been self reviewed
- [x] code runs without any errors
- [x] thorough testing has been implemented if adding feature
- [x] all tests pass

### Thanks!

### CODE REVIEW

File: lib/league_statistics.rb [hunk 1]
```diff
@@ -106,13 +105,12 @@ def highest_scoring_home_team
...
-      game_team[:hoa]== "home"
-      game_team[:team_id] == team_id
+      game_team[:hoa]== "home" && game_team[:team_id] == team_id
     end
     total_home_goals = home_games.map do |home_game|
-      home_game[:goals].to_i 
+      home_game[:goals].to_f
     end.sum
-    average_goals_per_hometeam = total_home_goals / home_games.length 
+    average_goals_per_hometeam = total_home_goals / home_games.length.to_f
```
Comment (kbug819) [inline lib/league_statistics.rb:113]: Great work team!

File: lib/league_statistics.rb [hunk 2]
```diff
@@ -66,9 +66,9 @@ def average_goals_for_team(team_id)
     end 
 
     total_goals = games.map do |game|
-      game[:goals].to_i 
+      game[:goals].to_f
```
Comment (elizakeating) [inline lib/league_statistics.rb:69]: Nice catch on changing this to float!

**Comment by kbug819** [comment]
Great, thank you for adding the methods! Loving to see these tests passing!

**Comment by elizakeating** [comment]
Woo! Almost there!

**Review by nurudeensulaimon — COMMENTED** [review]
Reviewed. Great catch on changing the data type for calculation to float!

**Pull request merged**